### ISSUE TITLE
Need an AngularJsUrlSerializer for router/upgrade

### ISSUE THREAD

**Issue by jasonaden** [angular/angular#21875]
### The Problem

Angular and AngularJS URL serialization differ in a couple key ways. One is the handling of special characters, and another is handling of empty URL params.

URL: `xyz.com?foo&bar=`

AngularJS parses as: `{foo: true, bar: ""}`
Angular parses as: `{foo: "", bar: ""}`

The result in the Angular router is:

`serializer.serialize(serializer.parse('xyz.com?foo&bar=')) == 'xyz.com?foo=&bar='`

Angular is adding in the `=` sign when re-serializing.

### Solution

Produce a new `AngularJsUrlSerializer`. This would be a serializer that passes all the parse/serialize tests in AngularJS. It wouldn't extend from the `DefaultUrlSerializer`, but would instead be totally new. It could be implemented similarly to the current `UrlParser` with some modifications, plus modifying the `serializeQueryParams` and `serializeParams` functions. We can test using this initial implementation to make sure it works as expected within Google before making available externally.

**Comment by jasonaden** [comment]
Closing this as it's no longer needed after #22337 gets merged.
